services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -0,0 +1,858 @@
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not update flow: Couldn't setup protected path for one-switch flow"
+
+        and: "Flow rules are still in multi table mode"
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            verifyAll(northbound.getSwitchRules(sw.dpId).flowEntries) { rules ->
+                rules.find { it.cookie == flowInfoFromDb.forwardPath.cookie.value }.tableId == 2
+                rules.find { it.cookie == flowInfoFromDb.reversePath.cookie.value }.tableId == 2
+            }
+        }
+
+        and: "Flow is valid and pingable"
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+        with(northbound.pingFlow(flow.flowId, new PingInput())) {
+            it.forward.pingSuccess
+            it.reverse.pingSuccess
+        }
+
+        and: "Cleanup: revert system to original state"
+        flowHelper.deleteFlow(flow.flowId)
+        northbound.updateSwitchProperties(sw.dpId, initSwProps)
+    }
+
+    def "Flow rules are not recreating while changing flow state to UP,DOWN"() {
+        given: "Three active switches"
+        List<PathNode> desiredPath = null
+        List<Switch> involvedSwitches = null
+        def switchPair = topologyHelper.allNotNeighboringSwitchPairs.collectMany { [it, it.reversed] }.find { pair ->
+            desiredPath = pair.paths.find { path ->
+                involvedSwitches = pathHelper.getInvolvedSwitches(path)
+                involvedSwitches.size() == 3
+            }
+        }
+        assumeTrue("Unable to find a path with three switches", switchPair.asBoolean())
+        //make required path the most preferred
+        switchPair.paths.findAll { it != desiredPath }.each { pathHelper.makePathMorePreferable(desiredPath, it) }
+        def initSrcSwProps = northbound.getSwitchProperties(involvedSwitches[0].dpId)
+        def initTransitSwProps = northbound.getSwitchProperties(involvedSwitches[1].dpId)
+        def initDstSwProps = northbound.getSwitchProperties(involvedSwitches[2].dpId)
+
+        and: "Multi table is disabled for them"
+        def multitableSrcSwIsDisabled = !initSrcSwProps.multiTable
+        def multitableTransitSwIsDisabled = !initTransitSwProps.multiTable
+        def multitableDstSwIsDisabled = !initDstSwProps.multiTable
+        multitableSrcSwIsDisabled ?: northbound.updateSwitchProperties(involvedSwitches[0].dpId,
+                northbound.getSwitchProperties(involvedSwitches[0].dpId).tap { it.multiTable = false })
+        multitableTransitSwIsDisabled ?: northbound.updateSwitchProperties(involvedSwitches[1].dpId,
+                northbound.getSwitchProperties(involvedSwitches[1].dpId).tap { it.multiTable = false })
+        multitableDstSwIsDisabled ?: northbound.updateSwitchProperties(involvedSwitches[2].dpId,
+                northbound.getSwitchProperties(involvedSwitches[2].dpId).tap { it.multiTable = false })
```

Comment:
all this is so bulky. Why no loops? what if there were 50 switches? can we just
```
def initSwProps = involvedSwitches.collectEntries { [(it.dpId): northbound.getSwitchProperties(it.dpId)]  }
involvedSwitches.findAll { initSwProps[it.dpId].multiTable }.each { 
    northbound.updateSwitchProperties(it.dpId, northbound.getSwitchProperties(it.dpId)
                    .tap { it.multiTable = false })
}
```

Same for cleanup actions. should be loop.
Applies to all tests here

Reply:
fixed everywhere